plugins/embed-optimizer/class-embed-optimizer-tag-visitor.php

```diff
@@ -214,21 +214,23 @@ private function reduce_layout_shifts( OD_Tag_Visitor_Context $context ): void {
 		 */
 		$minimums = array();
 
-		$denormalized_elements = $context->url_metric_group_collection->get_all_denormalized_elements()[ $embed_wrapper_xpath ] ?? array();
-		foreach ( $denormalized_elements as list( $group, $url_metric, $element ) ) {
-			if ( ! isset( $element['resizedBoundingClientRect'] ) ) {
+		$elements = $context->url_metric_group_collection->get_all_elements()[ $embed_wrapper_xpath ] ?? array();
+		foreach ( $elements as $element ) {
+			$resized_bounding_client_rect = $element->get( 'resizedBoundingClientRect' );
+			if ( ! is_array( $resized_bounding_client_rect ) ) {
 				continue;
 			}
+			$group           = $element->url_metric->group;
 			$group_min_width = $group->get_minimum_viewport_width();
 			if ( ! isset( $minimums[ $group_min_width ] ) ) {
 				$minimums[ $group_min_width ] = array(
 					'group'  => $group,
-					'height' => $element['resizedBoundingClientRect']['height'],
+					'height' => $resized_bounding_client_rect['height'],
 				);
 			} else {
 				$minimums[ $group_min_width ]['height'] = min(
 					$minimums[ $group_min_width ]['height'],
-					$element['resizedBoundingClientRect']['height']
+					$resized_bounding_client_rect['height']
```

Comment:
Just curious: Wouldn't all this marked code also have worked in the way it was before this change, given `OD_Element` implements `ArrayAccess`?

Reply:
Yes, it would have worked the same. I made this change because of a PHPStan issue where attempting to access `$element['resizedBoundingClientRect']['height']` results in a PHPStan error:

> phpstan: Cannot access offset 'height' on array{width: float, height: float, x: float, y: float, top: float, right: float, bottom: float, left: float}|bool|float|non-empty-string.

PHPStan isn't constraining the type based on the type of `$element['resizedBoundingClientRect']` based on the key `resizedBoundingClientRect` so the possible types are a union of all values in `ElementData`, which also doesn't necessarily include this extended `resizedBoundingClientRect` key's type (although here it is a DOMRect).

This goes back to wanting to be able to provide richer typing in PHPStan for getters like this, but in the meantime resorting to using `OD_Element::get()` which returns `mixed` and then checking the return value as being an array avoids the PHPStan error. 

Although I'm a bit surprised it's not complaining that the `$resized_bounding_client_rect['height']` key may not be defined. But this appears to be as expected: <https://phpstan.org/r/bc084e48-691a-4232-be15-7b806fda8b3e>. I recall there was some PHPStan extension to make checks even more strict than strict rules mandate all array keys be checked as well, but I can't find how to do that off hand.